src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java

```diff
@@ -87,22 +90,66 @@ public void handleRollFile(SourceFile fileDescriptor) {
          */
         for (int retryAttempts = 0; true; retryAttempts++) {
             try {
-                client.ingestFromFile(fileSourceInfo, ingestionProps);
+                IngestionResult ingestionResult = client.ingestFromFile(fileSourceInfo, ingestionProps.ingestionProperties);
+                if (ingestionProps.streaming && ingestionResult instanceof IngestionStatusResult) {
+                    // If IngestionStatusResult returned then the ingestion status is from streaming ingest -
+                    // Note: next version it will also contain Queued status - which is OK as getIngestionStatusCollection
+                    // won't cause any data fetch
+                    IngestionStatus ingestionStatus = ingestionResult.getIngestionStatusCollection().get(0);
+                    if (!hasStreamingSucceeded(ingestionStatus) && ingestionStatus.failureStatus.equals(IngestionFailureInfo.FailureStatusValue.Permanent)) {
+                       retryAttempts += ManagedStreamingIngestClient.MAX_RETRY_CALLS - 1;
```

Comment:
This is a little odd.
Isn't this equivalent of breaking out of the loop with some error handling?

Reply:
i think only misstake here is the check for permanent - ill remove it

Reply:
ok, but why are you moving the retry-counter all the way to the end?
What's the reason behind it

Reply:
oh ill add a comment about it (this max is coming from the client - the managed streaming does some retries by itself )
actually i dont think we would reach this code as we probably wont fail on queued fallback - maybe we should instead make the retries for streaming configureable